c_tests/tests/noopts.c

```diff
@@ -7,67 +7,52 @@
 //     ...
 //     store i32 2...
 //     ...
-//     store i32 3...
-//     ...
 //     ...add...
 //     ...
 //     --- End aot ---
 //     --- Begin jit-pre-opt ---
 //     ...
 //     store i32 2...
 //     ...
-//     store i32 3...
-//     ...
 //     ...add...
 //     ...
 //     --- End jit-pre-opt ---
 //     --- Begin jit-post-opt ---
 //     ...
 //     store i32 2...
 //     ...
-//     store i32 3...
-//     ...
 //     ...add...
 //     ...
 //     --- End jit-post-opt ---
 //     ...
 
-// Check that our NOOPT macro blocks optimisations.
+// Check that our NOOPT_VAL macro blocks optimisations.
 
 #include <assert.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
 #include <yk_testing.h>
 
-int add_them(int x, int y) {
-  // At the call-site x and y are constant, so an optimising compiler would try
-  // to evaluate `x + y` at compile time, replacing the result with a constant
-  // value. Our NOOPT macro should block this.
-  return x + y;
-}
-
 int main(int argc, char **argv) {
-  int x, y, res;
-  void *tt = __yktrace_start_tracing(HW_TRACING, &res, &x, &y);
-  // The __yktrace_start_tracing() call has almost certainly already blocked
-  // optimisation of x and y, as pointers to them are passed as arguments and
-  // the compiler must assume that it may mutate them. Therefore to properly
-  // test NOOPT we need initialise x and y after the call.
+  int x, res;
+  void *tt = __yktrace_start_tracing(HW_TRACING, &res, &x);
+  // __yktrace_start_tracing() will already inhibit optimisations on x since
+  // it's address is passed as an argument and the compiler must assume that it
+  // may be mutated. Therefore to properly test NOOPT_VAL we need initialise x
```

Comment:
s/we need initialise/we need to initialise/

[I also think that backticks around the variable names would make this easier to read!]